### PR TITLE
Added E2E tests for clicking the export button

### PR DESCRIPTION
**Pull request by malathi13**

### CODE REVIEW

File: ui/tests/e2e/test.js [hunk 1]
```diff
@@ -39,6 +39,36 @@ describe("End-to-end", () => {
     expect(grayDiv).toBeTruthy();
   });
 
+  test("Export to Saturn", async () => {
```
Comment (melissachang) [inline ui/tests/e2e/test.js]: opt: Change [beforeAll](https://jestjs.io/docs/en/api#beforeallfn-timeout) to [beforeEach](https://jestjs.io/docs/en/api#beforeeachfn-timeout), and remove the first two lines from your tests.

File: api/api.yaml [hunk 2]
```diff
@@ -87,6 +87,8 @@ definitions:
     properties:
       name:
         type: string
+      enableFieldSearch:
```
Comment (melissachang) [inline api/api.yaml]: Hmm, this diff shouldn't be here. Let's take a look when you're in the office

File: ui/tests/e2e/test.js [hunk 3]
```diff
@@ -39,6 +39,17 @@ describe("End-to-end", () => {
     expect(grayDiv).toBeTruthy();
   });
 
+  test("Export to Saturn With Facet", async () => {
```
Comment (melissachang) [inline ui/tests/e2e/test.js]: Let's do [this comment](https://github.com/DataBiosphere/data-explorer/pull/139#discussion_r214208134) so all tests start from a clean slate.

Comment (melissachang) [inline ui/tests/e2e/test.js]: Let's name the 2 tests:

- Export to Saturn - no selected cohort
- Export to Saturn - selected cohort

File: ui/tests/e2e/test.js (hunk 1 above)
Comment (malathi13) [reply ui/tests/e2e/test.js]: Oops, I did that in the VM and didnt do it locally.

File: ui/tests/e2e/test.js [hunk 4]
```diff
@@ -39,6 +39,36 @@ describe("End-to-end", () => {
     expect(grayDiv).toBeTruthy();
   });
 
+  test("Export to Saturn", async () => {
+    await page.goto("http://localhost:4400");
+    await page.waitForSelector("span.datasetName");
+    await Promise.all([
```
[the remark this answers is not in the document]
Comment (malathi13) [reply ui/tests/e2e/test.js]: I think the correct way to specify an attribute selector is:
element[attr=value]
https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors

So it should actually be page.click("button[title='Send to Saturn']")

File: ui/tests/e2e/test.js [hunk 5]
```diff
@@ -39,6 +39,36 @@ describe("End-to-end", () => {
...
+    // Wait for data to be returned from backend.
+    // See #63 for why we can't wait for div.grayText.
+    await page.waitForXPath(
+      "//div[contains(@class, 'totalCountText') and text() = '1018']"
+    );
+    await page.click("button", { title: "Send to Saturn" });
+    await page.waitForSelector("#name");
+
+    await page.type("#name", "test-cohort");
+    await Promise.all([page.click("#save"), page.waitFor(15000)]);
```
[the remark this answers is not in the document]
Comment (malathi13) [reply ui/tests/e2e/test.js]: I think it's better to wait for 10-15 secs to allow the bvdp page to load, checking for the url instantly might be flaky.

File: ui/tests/e2e/test.js [hunk 6]
```diff
@@ -39,6 +39,36 @@ describe("End-to-end", () => {
...
+    await page.goto("http://localhost:4400");
+    await page.waitForSelector("span.datasetName");
+    await Promise.all([
+      page.click("button[title='Send to Saturn']"),
+      page.waitFor(15000)
+    ]);
+    expect(await page.url()).toBe("https://bvdp-saturn-prod.appspot.com/");
+  });
+
+  test("Export to Saturn With Facet", async () => {
```
[the remark this answers is not in the document]
Comment (malathi13) [reply ui/tests/e2e/test.js]: done

File: ui/tests/e2e/test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (malathi13) [reply ui/tests/e2e/test.js]: done

File: docker-compose.yml [hunk 7]
```diff
@@ -38,6 +38,7 @@ services:
     # --reload will restart the server after code change
     command: ["-b", ":8390", "--reload"]
     environment:
+      - GOOGLE_APPLICATION_CREDENTIALS
```
[the remark this answers is not in the document]
Comment (malathi13) [reply docker-compose.yml]: done

File: .circleci/config.yml [hunk 8]
```diff
@@ -44,19 +44,15 @@ jobs:
...
               tox
       - run:
           name: Run docker-compose
           command: |
-              # Remove deploy.json since it doesn't contain project_id.
-              # With no deploy.json, API server will print a warning but will
-              # otherwise run normally.
-              rm dataset_config/1000_genomes/deploy.json
-              docker-compose up --build -d
+              GOOGLE_APPLICATION_CREDENTIALS="dataset_config/1000_genomes/private-key.json" docker-compose up --build -d
```
[the remark this answers is not in the document]
Comment (malathi13) [reply .circleci/config.yml]: done

File: .circleci/config.yml [hunk 9]
```diff
@@ -44,19 +44,15 @@ jobs:
           name: Run API server unit tests
           command: |
               # Some files are missing because of .gitignore
-              touch dataset_config/1000_genomes/deploy.json
-              touch dataset_config/1000_genomes/private-key.json
+              echo "{ \"project_id\": \"ci-test-data-explorer\"}" > dataset_config/1000_genomes/deploy.json
```
[the remark this answers is not in the document]
Comment (malathi13) [reply .circleci/config.yml]: done

File: ui/tests/e2e/test.js [hunk 10]
```diff
@@ -39,6 +39,31 @@ describe("End-to-end", () => {
     expect(grayDiv).toBeTruthy();
   });
 
+  test("Export to Saturn - no selected cohort", async () => {
+    await Promise.all([
```
Comment (melissachang) [inline ui/tests/e2e/test.js]: Make sense, can you put 43-46 on one line, like the other test?

File: ui/tests/e2e/test.js [hunk 11]
```diff
@@ -39,6 +39,31 @@ describe("End-to-end", () => {
...
+  test("Export to Saturn - selected cohort", async () => {
+    // Click first Super Population facet value.
+    let facetValueRow = await getFacetValueRow("Super Population", "African");
+    await facetValueRow.click("input");
+    // Wait for data to be returned from backend.
+    // See #63 for why we can't wait for div.grayText.
+    await page.waitForXPath(
+      "//div[contains(@class, 'totalCountText') and text() = '1018']"
+    );
+    await page.click("button[title='Send to Saturn']"),
```
Comment (melissachang) [inline ui/tests/e2e/test.js]: 59-60 look strange, how about
```
await page.click("button[title='Send to Saturn']")
await page.waitForSelector("#name");
```

**Pull request merged**